### PR TITLE
fix(bigquery-jdbc): refine temporal timezone coercion and PreparedStatement parameter setters

### PR DESCRIPTION
**Pull request by Neenu1995**
b/535194644

* **Centralized Timezone Coercion**: Consolidated `Calendar` timezone conversions in `BigQueryTypeCoercionUtility` using `java.time` APIs.
* **Temporal Helper Refactoring**: Extracted `setTemporalObject` helper to eliminate duplicate temporal type logic across `setObject` overloads.
* **Defensive Copying & Unsupported Stubs**: Added defensive copying for `Calendar` setters and replaced silent stubs with `BigQueryJdbcSqlFeatureNotSupportedException`.
* **Expanded Test Suite**: Added unit and integration tests covering temporal setters and calendar conversion edge cases.

### CODE REVIEW

File: java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryTypeCoercionUtility.java [hunk 1]
```diff
@@ -43,6 +46,98 @@ class BigQueryTypeCoercionUtility {
...
+    Calendar defaultCal = Calendar.getInstance();
+    defaultCal.setTime(time);
+
+    Calendar targetCal = getSafeCalendar(cal);
+    targetCal.set(Calendar.HOUR_OF_DAY, defaultCal.get(Calendar.HOUR_OF_DAY));
+    targetCal.set(Calendar.MINUTE, defaultCal.get(Calendar.MINUTE));
+    targetCal.set(Calendar.SECOND, defaultCal.get(Calendar.SECOND));
+    targetCal.set(Calendar.MILLISECOND, defaultCal.get(Calendar.MILLISECOND));
+    return new Time(targetCal.getTimeInMillis());
+  }
```
[the remark this answers is not in the document]
Comment (Neenu1995) [reply java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryTypeCoercionUtility.java:119]: Has to be Calendar. Or DST is not handled properly.

File: java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java [hunk 2]
```diff
@@ -470,24 +510,42 @@ public void setArray(int parameterIndex, Array x) throws SQLException {
...
+    }
+    setTime(parameterIndex, BigQueryTypeCoercionUtility.convertTimeWithCalendar(x, cal));
   }
 
   @Override
-  public void setTimestamp(int parameterIndex, Timestamp x, Calendar cal) {
-    // TODO :NOT IMPLEMENTED
+  public void setTimestamp(int parameterIndex, Timestamp x, Calendar cal) throws SQLException {
+    checkClosed();
+    if (x == null) {
```
Comment (logachev) [inline java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java:544]: Do we need this extra check? `convertXWithCalendar` will return `null` if `x==null`

File: java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java [hunk 3]
```diff
@@ -70,7 +76,7 @@ class BigQueryPreparedStatement extends BigQueryStatement implements PreparedSta
   protected int parameterCount = 0;
   protected String currentQuery;
   private Queue<ArrayList<BigQueryJdbcParameter>> batchParameters = new LinkedList<>();
-  private Schema insertSchema = null;
+  Schema insertSchema = null;
```
Comment (logachev) [inline java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java:79]: Why changing this? I don't see  where that's used

**Review by logachev — APPROVED** [review]

File: java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java (hunk 2 above)
Comment (Neenu1995) [reply java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java:544]: The setParameter method directly calls a toString method without a null check. If we don't check null here, it will cause an NPE.

File: java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java (hunk 3 above)
Comment (Neenu1995) [reply java-bigquery-jdbc/src/main/java/com/google/cloud/bigquery/jdbc/BigQueryPreparedStatement.java:79]: This was changed for unit test access.
